typesystem/types.h

```diff
@@ -62,6 +62,18 @@ struct CurrentVariant : CurrentSuper {};
 
 #define IS_EMPTY_CURRENT_STRUCT(T) (::current::reflection::IsEmptyCurrentStruct<T, IS_CURRENT_STRUCT(T)>::value)
 
+// The `_CURRENT_` part of the name is only here to not pollute the "global macros namespace".
+// The macro just checks whether `T` is an `std::tuple<...>`.
+template <typename>
+struct CurrentTupleDetector { constexpr static bool value = false; };
+template <typename... TS>
+struct CurrentTupleDetector<std::tuple<TS...>> { constexpr static bool value = true; };
+#define IS_CURRENT_TUPLE(T) (::crnt::CurrentTupleDetector<::current::decay<T>>::value)
```

Comment:
There's no such thing as "Current tuple", right? Maybe just add `is_tuple` to `tuple.h`?